### DISCUSSION TITLE
Cleaning compound shape

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#195] [Q&A] [answered]
If I `shape.RemoveShape(n)` from a mutable compound shape, do I need to clean/destroy the child shape that was removed? I tried to `shape.Release()`, but then it breaks my debug drawer, which tries to

```
const shape = body.GetShape();
const triContext = new Jolt.ShapeGetTriangles(shape, ...);
```

**Comment by jrouwe** [comment, marked as answer]
In principle you don't need to call `Release` (will happen internally), unless you did an `AddRef` yourself.
